gobblin-api/src/main/java/gobblin/fork/CopyHelper.java

```diff
@@ -56,37 +56,26 @@ private static boolean isImmutableType(Object thing) {
 
   /**
    * Copy this object if needed.
-   * @param thing: this object that needs to be copied
-   * @param instance: the number of instances of this object already copied
+   * @param thing : this object that needs to be copied
    * @return: a possibly copied instance
    * @throws CopyNotSupportedException if thing needs to be copied but cannot be
    */
-  public static Object copy(Object thing, int instance) throws CopyNotSupportedException {
-    // First copy is special, we only copy if the thing is an instance of Copyable
```

Comment:
Because the caller doesn't need to keep track of the instance number, just whether the object needs to be copied or not. For mutable, trivially copyable objects, like byte array, it is incorrect to return `thing` here, as the first fork may mutate the object before it gets copied into other forks (race condition). 

The only instance that is safe to copy is the last one, but then we have an implicit contract that the instances are passed in order to the Copier, which is not specified anywhere and is hard to enforce.